samtranslator/third_party/py27hash/hash.py

```diff
@@ -2,11 +2,25 @@
 Compatibility methods to support Python 2.7 style hashing in Python 3.X+
 
 This is designed for compatibility not performance.
+
 """
 
 import ctypes
 import math
 
+# See https://github.com/python/mypy/issues/5107
+from typing import TYPE_CHECKING, TypeVar, Callable, Any
+
+if TYPE_CHECKING:
+    F = TypeVar("F", bound=Callable[[Any], Any])
+
+    def lru_cache(maxsize: int = 128, typed: bool = False) -> Callable[[F], F]:
+        pass
+
+
+else:
```

Comment:
Any benefit versus `# type: ignoring` with a comment?

Reply:
I was probably doing it wrong, but I was unable to make `# type:ignoring` work; I still got complaints about unused type annotations as in original commit.

Reply:
Ah ok :( Even when you put `# type: ignore` on the exact line that it complains about?